meta/saimetadatautils.c

```diff
@@ -150,6 +150,75 @@ const sai_attr_metadata_t* sai_metadata_get_attr_metadata_by_attr_id_name(
     return NULL;
 }
 
+static int sai_metadata_attr_id_name_cmp(
+        _In_ const char * str1,
+        _In_ const char * str2)
+{
+    char c1 = 0;
+    char c2 = 0;
+
+    while (true)
+    {
+        c1 = *str1++;
+        c2 = *str2++;
+
+        if (sai_serialize_is_char_allowed(c1) || sai_serialize_is_char_allowed(c2) || c1 != c2)
+        {
+            if (sai_serialize_is_char_allowed(c1))
+            {
+                c1 = 0;
+            }
+
+            if (sai_serialize_is_char_allowed(c2))
+            {
+                c2 = 0;
+            }
+
+            return c1 - c2;
+        }
+    }
+}
+
+const sai_attr_metadata_t* sai_metadata_get_attr_metadata_by_attr_id_name_ext(
```

Comment:
please add test in saisanitycheck.c that will compare results from sai_metadata_get_attr_metadata_by_attr_id_name_ext and sai_metadata_get_attr_metadata_by_attr_id_name, iterating via all attributes to see if the results will be the same

Reply:
ACK.